src/System.Runtime.Extensions/src/System/Diagnostics/Stopwatch.Windows.cs

```diff
@@ -6,14 +6,24 @@ namespace System.Diagnostics
 {
     public partial class Stopwatch
     {
-        private static bool QueryPerformanceFrequency(out long value)
+        private static unsafe long QueryPerformanceFrequency()
         {
-            return Interop.Kernel32.QueryPerformanceFrequency(out value);
+            long resolution;
+
+            Interop.BOOL result = Interop.Kernel32.QueryPerformanceFrequency(&resolution);
+            Debug.Assert(result != Interop.BOOL.FALSE);
+
+            return resolution;
         }
 
-        private static bool QueryPerformanceCounter(out long value)
+        private static unsafe long QueryPerformanceCounter(long* lpPerformanceCount)
         {
-            return Interop.Kernel32.QueryPerformanceCounter(out value);
+            long timestamp;
+
+            Interop.BOOL result = Interop.Kernel32.QueryPerformanceCounter(&timestamp);
+            Debug.Assert(result != Interop.BOOL.FALSE);
```

Comment:
Why can't this fail?  If it does, we're going to end up with garbage in timestamp.  Same for above with resolution.

Reply:
This function is documented to always succeed on Windows XP and newer: https://msdn.microsoft.com/en-us/library/windows/desktop/ms644904(v=vs.85).aspx

The error handling here was a left over from Windows 95 and friends.

Reply:
Great.

Reply:
Maybe useful to add a short comment "This API is documented to never fail"

Reply:
Fixed.